sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Common/src/Shared/Queues/QueueMetricsProvider.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Storage.Common.Listeners
+{
+    /// <summary>
+    /// Provides QueueTriggerMetrics from a specific queue entity.
+    /// </summary>
+    internal class QueueMetricsProvider
+    {
+        private readonly QueueClient _queue;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Instantiates a QueueMetricsProvider.
+        /// </summary>
+        /// <param name="queue">The QueueClient to use for metrics polling.</param>
+        /// <param name="loggerFactory">Used to create an ILogger instance.</param>
+        public QueueMetricsProvider(QueueClient queue, ILoggerFactory loggerFactory)
+        {
+            _queue = queue;
+            _logger = loggerFactory.CreateLogger<QueueMetricsProvider>();
+        }
+
+        // Caller of this function is responsible for wrapping this in try/catch
+        public async Task<int> GetQueueLength()
+        {
+            QueueProperties queueProperties = await _queue.GetPropertiesAsync().ConfigureAwait(false);
+            return queueProperties.ApproximateMessagesCount;
+        }
+
+        /// <summary>
+        /// Retrieves metrics from the queue entity.
+        /// </summary>
+        /// <returns></returns>
```

Comment:
Add return description as the summary exists

Reply:
done